libs/wingsdk/test/global.setup.ts

```diff
@@ -1,8 +1,86 @@
-import { exec, execSync } from "child_process";
+import { execSync } from "child_process";
+import fs from "fs/promises";
+import path from "path";
 
-export default async function () {
+export async function setup() {
   // compile src/**/*.on*.inflight.ts to .js because these are going to be
   // injected into our javascript vm and cannot be resolved via vitest
   execSync("npx tsc -p tsconfig.test.json");
   return () => {};
 }
+
+// remove all the compiled .js and .d.ts files after the tests are done
+// so that we don't have a ocean of files to sift through
+export async function teardown() {
```

Comment:
I worry this adds a little bit of fragility/wasted time. I think it would be worth exploring a way to compile these files to a different directory and making sure vitest resolves it from there.

Reply:
@MarkMcCulloh I get the concern, I played around with vitest resolve aliases for a bit last night but did not have any success getting tests to pass. I assume it has to do with the way the inflight client requires do not get the the replacement applied. (at least that is what my testing seemed to reveal)

Im fine passing on this or bringing it in as a temp solution. I just think the generated mess right now is cumbersome.